document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresDatastore.java

```diff
@@ -26,47 +26,19 @@ public class PostgresDatastore implements Datastore {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(PostgresDatastore.class);
 
-  private static final String DEFAULT_USER = "postgres";
-  private static final String DEFAULT_PASSWORD = "postgres";
-  private static final String DEFAULT_DB_NAME = "postgres";
-  private static final int DEFAULT_MAX_CONNECTION_ATTEMPTS = 200;
-  private static final Duration DEFAULT_CONNECTION_RETRY_BACKOFF = Duration.ofSeconds(5);
-
   private PostgresClient client;
   private String database;
+  private PostgresConnectionPool connectionPool;
 
   @Override
   public boolean init(Config config) {
     try {
       DriverManager.registerDriver(new org.postgresql.Driver());
-      this.database = config.hasPath("database") ? config.getString("database") : DEFAULT_DB_NAME;
-
-      String url;
-      if (config.hasPath("url")) {
-        url = config.getString("url");
-      } else {
-        String hostName = config.getString("host");
-        int port = config.getInt("port");
-        url = String.format("jdbc:postgresql://%s:%s/", hostName, port);
-      }
-
-      String user = config.hasPath("user") ? config.getString("user") : DEFAULT_USER;
-      String password =
-          config.hasPath("password") ? config.getString("password") : DEFAULT_PASSWORD;
-      int maxConnectionAttempts =
-          config.hasPath("maxConnectionAttempts")
-              ? config.getInt("maxConnectionAttempts")
-              : DEFAULT_MAX_CONNECTION_ATTEMPTS;
-      Duration connectionRetryBackoff =
-          config.hasPath("connectionRetryBackoff")
-              ? config.getDuration("connectionRetryBackoff")
-              : DEFAULT_CONNECTION_RETRY_BACKOFF;
-
-      String finalUrl = url + this.database;
-      client =
-          new PostgresClient(
-              finalUrl, user, password, maxConnectionAttempts, connectionRetryBackoff);
+      final PostgresConfig postgresConfig = ConfigBeanFactory.create(config, PostgresConfig.class);
 
+      client = new PostgresClient(postgresConfig);
+      database = postgresConfig.getDatabase();
+      connectionPool = new PostgresConnectionPool(postgresConfig);
```

Comment:
We don't have to create `connectionPool` here, right? Its already created as part of `PostgresClient` and its accessed via getPooledConnection.

Reply:
Good catch! Updated.